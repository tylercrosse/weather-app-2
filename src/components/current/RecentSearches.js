import React from "react";
import PropTypes from "prop-types";

class RecentSearches extends React.Component {
  constructor(props) {
    super(props);

    this.handleClick = this.handleClick.bind(this);
  }
  handleClick(location) {
    this.props.fetchForecast(location);
  }
  render() {
    const { locations } = this.props;
    const recentSearchItems = Object.values(locations)
      .sort((a, b) => b.time - a.time)
      .slice(0, 3)
      .map(location => {
        return (
          <button
            key={location.id}
            className="recentSearch__item"
            onClick={() => this.handleClick(location)}
          >
            <svg
              xmlns="http://www.w3.org/2000/svg"
              viewBox="0 0 48 48"
            >
              <path d="M24,0A24,24,0,1,0,48,24,24,24,0,0,0,24,0Zm0,42A18,18,0,1,1,42,24,18,18,0,0,1,24,42Zm9-21H27V15a3,3,0,0,0-6,0v9a3,3,0,0,0,3,3h9a3,3,0,0,0,0-6Z" />
            </svg>
            <p>{location.address}</p>
          </button>
        );
      });
    return (
      <div className="recentSearches">
        {recentSearchItems}
      </div>
    );
  }
}

RecentSearches.propTypes = {
  locations: PropTypes.object.isRequired,
  fetchForecast: PropTypes.func.isRequired
};

export default RecentSearches;
