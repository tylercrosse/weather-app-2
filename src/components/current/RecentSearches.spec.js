import React from 'react';
import { mount } from 'enzyme';
import toJson from "enzyme-to-json";
import RecentSearches from './RecentSearches';

const setup = () => {
  const props = {
    locations: {
      ChIJVTPokywQkFQRmtVEaUZlJRA: {
        time: 1499098513,
        id: 'ChIJVTPokywQkFQRmtVEaUZlJRA',
        address: 'Seattle, WA, USA',
        lat: 47.6062095,
        lng: -122.3320708
      },
      'ChIJW-T2Wt7Gt4kRKl2I1CJFUsI': {
        time: 1499061460,
        id: 'ChIJW-T2Wt7Gt4kRKl2I1CJFUsI',
        address: 'Washington, DC, USA',
        lat: 38.9071923,
        lng: -77.03687070000001
      },
    },
    fetchForecast: jest.fn(),
  }

  const component = <RecentSearches {...props} />;
  const wrapper = mount(component);

  return {
    props,
    wrapper
  }
}

describe('<RecentSearches />', () => {
  it('should render correctly', () => {
    const { wrapper } = setup();
    expect(toJson(wrapper)).toMatchSnapshot();
  });
})
