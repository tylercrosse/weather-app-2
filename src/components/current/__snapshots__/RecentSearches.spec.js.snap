// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecentSearches /> should render correctly 1`] = `
<RecentSearches
  fetchForecast={[Function]}
  locations={
    Object {
      "ChIJVTPokywQkFQRmtVEaUZlJRA": Object {
        "address": "Seattle, WA, USA",
        "id": "ChIJVTPokywQkFQRmtVEaUZlJRA",
        "lat": 47.6062095,
        "lng": -122.3320708,
        "time": 1499098513,
      },
      "ChIJW-T2Wt7Gt4kRKl2I1CJFUsI": Object {
        "address": "Washington, DC, USA",
        "id": "ChIJW-T2Wt7Gt4kRKl2I1CJFUsI",
        "lat": 38.9071923,
        "lng": -77.03687070000001,
        "time": 1499061460,
      },
    }
  }
>
  <div
    className="recentSearches"
  >
    <button
      className="recentSearch__item"
      onClick={[Function]}
    >
      <svg
        viewBox="0 0 48 48"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M24,0A24,24,0,1,0,48,24,24,24,0,0,0,24,0Zm0,42A18,18,0,1,1,42,24,18,18,0,0,1,24,42Zm9-21H27V15a3,3,0,0,0-6,0v9a3,3,0,0,0,3,3h9a3,3,0,0,0,0-6Z"
        />
      </svg>
      <p>
        Seattle, WA, USA
      </p>
    </button>
    <button
      className="recentSearch__item"
      onClick={[Function]}
    >
      <svg
        viewBox="0 0 48 48"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M24,0A24,24,0,1,0,48,24,24,24,0,0,0,24,0Zm0,42A18,18,0,1,1,42,24,18,18,0,0,1,24,42Zm9-21H27V15a3,3,0,0,0-6,0v9a3,3,0,0,0,3,3h9a3,3,0,0,0,0-6Z"
        />
      </svg>
      <p>
        Washington, DC, USA
      </p>
    </button>
  </div>
</RecentSearches>
`;
